nvaccess/nvda
# Braille: fix auto-tether after moving mouse and using review commands, allow panning after review commands

### Link to issue number:
Fixes #8682
Fixes #11519
### Summary of the issue:
#8682: when using the read_currentLine, read_currentWord, and read_currentCharacter commands after the caret has been moved, it is impossible to pan the Braille display through the displayed content.
#11519: when moving the mouse with Mouse Tracking on (default) and Braille Tether set to Auto (default), using read_currentLine, read_currentWord, and read_currentCharacter speaks content under the mouse, but does not update the Braille display with the spoken content. 
### Description of how this pull request fixes the issue:
These three global review functions all call braille.handler.setTether early in their processing. braille.handler.setTether clears the display buffer, making it impossible to scroll through the displayed content. braille.handler.setTether also does not update the display, causing the problems discussed in #11519 -- tether flags are set properly, but the display does not change. A simple call to braille.handler.handleReviewMove in all three cases solves both of these issues.
### Testing strategy:
In NVDA settings, be sure Enable mouse tracking is checked, and Tether Braille is set to Automatically.
1. Connect to a Braille display.
2. Load a file in Notepad.
3. Move the mouse to a section of the file other than the section containing the caret.
4. Press numpad8, Report current line in review.
5. Braille display should show content under the mouse instead of remaining unchanged.
6. Repeat steps 2 through 5, but use numpad5 and numpad2 in the test.
7. Repeat steps 2 through 5, but attempt to pan the display through content immediately after pressing numpad8, numpad5, and numpad2. Content should pan normally.
### Known issues with pull request:
None observed thus far.
### Change log entries:
Bug fixes
* When Braille is tethered automatically and the mouse is moved with Enable mouse tracking on, the Report current line in review, Report current word in review, and Report current character in review commands now update the Braille display with the spoken content.

* It is now possible to pan the Braille display through content after use of the Report current line in review, Report current word in review, and Report current character in review commands.
### Code Review Checklist:



  - description is up to date
  - change log entries
  - Unit tests
  - System (end to end) tests
  - Manual testing
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
